.github/workflows/deploy.yml

```diff
@@ -183,3 +183,56 @@ jobs:
       - name: Check that batches are being verified
         working-directory: ./scripts
         run: ./batch_verification_monitor.sh ${{ env.BATCH_VERIFICATION_MONITOR_TARGET }} ${{ env.BATCH_VERIFICATION_MONITOR_TIMEOUT }}
+
+  # Deploy the CDK environment against a local l1 chain with pre-deployed zkevm contracts.
+  pre-deployed-contracts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      # Install tools.
+      - name: Install kurtosis
+        run: |
+          echo "deb [trusted=yes] https://apt.fury.io/kurtosis-tech/ /" | sudo tee /etc/apt/sources.list.d/kurtosis.list
+          sudo apt update
+          sudo apt install kurtosis-cli=${{ env.KURTOSIS_VERSION }}
+          kurtosis analytics disable
+
+      - name: Install yq
+        run: pip3 install yq
+
+      - name: Install foundry
+        uses: foundry-rs/foundry-toolchain@v1
+
+      # Deploy components.
+      - name: Deploy L1 and zkEVM contracts
+        run: |
+          # Disable all deployment steps.
+          yq -Y --in-place 'with_entries(if .value | type == "boolean" then .value = false else . end)' params.yml
+          # Only deploy l1 and zkevm contracts.
+          yq -Y --in-place '(.deploy_l1 = true) | (.deploy_zkevm_contracts_on_l1 = true)' params.yml
+
+          kurtosis run --enclave cdk-v1 --args-file params.yml --image-download always .
+
+      - name: Specify the rollup manager contract address and a few other parameters
+        run: |
+          zkevm_rollup_manager_address=$(kurtosis service exec cdk-v1 contracts-001 "cat /opt/zkevm/combined.json" | tail -n +2 | jq .polygonRollupManagerAddress)
```

Comment:
nit: any way we can get service name (i.e. contracts-001) programatically from kurtosis instead of hardcoding?

Reply:
I think we could try but I don't really see the point. It's already hardcoded in the code and it should not be changed by users so we can rely on the assumption that this service will always be called `contracts-{suffix}`.